Cubical/Data/Quiver/Base.agda

```diff
@@ -58,66 +59,59 @@ Quiver : ∀ ℓg ℓg' → Type _
 Quiver ℓg ℓg' = Σ[ ob ∈ Type ℓg ] QuiverOver ob ℓg'
 
 -- A "heteromorphism" from a Quiver to a Graph
-record HetQG (Q : Quiver ℓc ℓc') (G : Graph ℓg ℓg')
-       : Type (ℓ-max (ℓ-max ℓc ℓc') (ℓ-max ℓg ℓg')) where
+record HetQG (Q : Quiver ℓq ℓq') (G : Graph ℓg ℓg')
+       : Type (ℓ-max (ℓ-max ℓq ℓq') (ℓ-max ℓg ℓg')) where
   field
     _$g_ : Q .fst → G .Node
     _<$g>_ : (e : Q .snd .mor)
            → G .Edge (_$g_ (Q .snd .dom e)) (_$g_ (Q .snd .cod e))
 open HetQG public
 
-Graph→Quiver : Graph ℓc ℓc' → Quiver ℓc (ℓ-max ℓc ℓc')
+module _ {G : Graph ℓg ℓg'}
+         {H : Graph ℓh ℓh'}
+         {Q : Quiver ℓq ℓq'}
+         where
+  compGrHomHetQG : GraphHom G H → HetQG Q G → HetQG Q H
+  compGrHomHetQG ϕ h ._$g_   q = ϕ GraphHom.$g (h HetQG.$g q)
+  compGrHomHetQG ϕ h ._<$g>_ e = ϕ GraphHom.<$g> (h HetQG.<$g> e)
+
+-- Universal property:
+-- HetQG Q G ≅ QHom Q (Graph→Quiver G)
+Graph→Quiver : Graph ℓg ℓg' → Quiver ℓg (ℓ-max ℓg ℓg')
 Graph→Quiver g .fst = g .Node
 Graph→Quiver g .snd .mor = Σ[ A ∈ g .Node ] Σ[ B ∈ g .Node ] g .Edge A B
 Graph→Quiver g .snd .dom x = x .fst
 Graph→Quiver g .snd .cod x = x .snd .fst
 
 -- | The use of ≡ in this definition is the raison d'etre for the
 -- | Quiver-Graph distinction
-Quiver→Graph : Quiver ℓc ℓc' → Graph ℓc (ℓ-max ℓc ℓc')
+-- HetQG Q G ≅ QHom (Quiver→Graph Q) G
+Quiver→Graph : Quiver ℓq ℓq' → Graph _ _
 Quiver→Graph Q .Node = Q .fst
 Quiver→Graph Q .Edge A B =
   Σ[ e ∈ Q .snd .mor ]
     (Q .snd .dom e ≡ A)
   × (Q .snd .cod e ≡ B)
 
-Cat→Quiver : Category ℓc ℓc' → Quiver ℓc (ℓ-max ℓc ℓc')
+Cat→Quiver : Category ℓC ℓC' → Quiver _ _
 Cat→Quiver 𝓒 = Graph→Quiver (Cat→Graph 𝓒)
 
-module _ (Q : Quiver ℓg ℓg') where
-  -- Of the four possible combinations: Q over Q, G over Q, Q over G,
-  -- Q over Q, this strikes the right balance. Having the displayed
-  -- thing be a Graph means we can define sections without using ≡.
-  record Graphᴰ ℓgᴰ ℓgᴰ' : Type (ℓ-max (ℓ-max ℓg ℓg') (ℓ-suc (ℓ-max ℓgᴰ ℓgᴰ')))
+-- A "heterogeneous" local section of a displayed graph
```

Comment:
heterogenous in what sense?

Reply:
bc it's a graph over a quiver rather than a quiver over a quiver or graph over a graph